src/io/import/processors/updateUriType.ts

```diff
@@ -0,0 +1,55 @@
+import StreamingByteReader from '@/src/core/streaming/streamingByteReader';
+import { ImportHandler } from '@/src/io/import/common';
+import { getFileMimeFromMagicStream } from '@/src/io/magic';
+import { asCoroutine } from '@/src/utils';
+
+const DoneSignal = Symbol('DoneSignal');
+
+function detectStreamType(stream: ReadableStream) {
+  return new Promise<string>((resolve, reject) => {
+    const reader = new StreamingByteReader();
+    const consume = asCoroutine(getFileMimeFromMagicStream(reader));
+
+    const writableStream = new WritableStream({
+      write(chunk) {
+        const result = consume(chunk);
+        if (result.done) {
+          const mime = result.value;
+          resolve(mime ?? '');
+          throw DoneSignal;
+        }
+      },
+    });
```

Comment:
When I `npx http-serve` a directory with a "remote manifest JSON" file, `result.done` is never true and this promise never resolves. The `close` Stream callback is called though.

Example remote manifest JSON
```
{
  "resources": [
    {
      "url": "http://localhost:9999/3DUS-Fetus.mha",
      "name": "3DUS-Fetus.mha"
    }
  ]
}
```

Reply:
I've rebased and added a commit that fixes this issue.